As a developer, I want to update si-units and indriya to 2.2.2 and 2.2.3 or later respectively

See #510. I attempted these updates:

```
 - si.uom:si-units [2.1 -> 2.2.2]
     https://unitsofmeasurement.github.io/
 - tech.units:indriya [2.2.2 -> 2.2.3]
     https://unitsofmeasurement.github.io/indriya/
```

I got this build error:

```
java.lang.NoClassDefFoundError: Could not initialize class wres.datamodel.units.Units
      at wres.pipeline.statistics.StatisticsProcessorTest.ofMetricProcessorForSingleValuedPairs(StatisticsProcessorTest.java:141)
  Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.NoSuchFieldError: UNIFIED_ATOMIC_MASS [in thread "Test worker"]
```

This might be a low-hanging fruit. I'm not sure.

Hank

--- a/build.gradle
+++ b/build.gradle
@@ -349,7 +349,7 @@ project(':wres-datamodel') {
 
         // The following five libraries together provide units of measure.
         api 'javax.measure:unit-api:2.2'
-        runtimeOnly 'si.uom:si-units:2.1'
+        runtimeOnly 'si.uom:si-units:2.2.2'
         implementation 'tech.units:indriya:2.2.3'
         implementation 'systems.uom:systems-ucum:2.2'
         implementation 'com.google.protobuf:protobuf-java:3.25.6'
@@ -376,7 +376,7 @@ project(':wres-datamodel') {
         testRuntimeOnly 'com.google.protobuf:protobuf-java:3.25.6'
         testImplementation 'org.hamcrest:hamcrest-core:1.3'
         testRuntimeOnly 'si.uom:si-quantity:2.1'
-        testImplementation 'si.uom:si-units:2.1'
+        testImplementation 'si.uom:si-units:2.2.2'
     }
     // Facilitate unit tests with JUnit
     test {

